### ISSUE TITLE
Some components' labels no longer focus on the control

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#5181]
Following #4831, it's no longer possible to click on for example the "Dropdown Date" label and get the focus on the first of the three dropdowns. Nothing happens.

**Comment by avernet** [comment]
Just with HTML, it seems that the browser only does this if there is a `<label for="…">` pointing to a field. But for those compound components, doing so would be incorrect, as that label is for a set of fields, not for the first field. So I think that to fix this we also need to resort to some JavaScript, doing the following:

1. Instead of using `<xh:div role="group" aria-label="{xxf:label('fr-xxx-binding')}">`, put an id on that container, and `xxbl:label-for="my-group"`. This is better anyway, as it avoids duplicating the label in the DOM (cleaner), and solves the problem of the label being in plain text but containing tags. (Not sure why I hadn't thought of doing this. Could it be that this didn't work in some earlier version of Orbeon Forms?)
2. Add some generic JavaScript that upon click on a `<label>` without `for`, finds if there is an element with `role="group"` and an `aria-label="…"` pointing to the label, and if so, sets the focus on the first form field inside that group.

**Comment by avernet** [comment]
Ah! Item 1 above doesn't work, unfortunately: `for` can only point to form controls, so we can't point from a label to the id of a group, and instead need to either do it the other way around (point from group to the label with an `aria-labelledby`), or use `aria-label` on the group (which is what we are doing right now).

**Comment by ebruchez** [comment]
With #5367 item 1 should now work.

But moving this to 2022.1.1.
